src/esp/sensor/PinholeCamera.cpp

```diff
@@ -28,8 +29,37 @@ void PinholeCamera::setProjectionParameters(SensorSpec::ptr spec) {
   hfov_ = std::atof(spec_->parameters.at("hfov").c_str());
 }
 
-void PinholeCamera::setProjectionMatrix(gfx::RenderCamera& targetCamera) {
+PinholeCamera& PinholeCamera::setProjectionMatrix(
+    gfx::RenderCamera& targetCamera) {
   targetCamera.setProjectionMatrix(width_, height_, near_, far_, hfov_);
+  return *this;
+}
+
+PinholeCamera& PinholeCamera::setModelViewMatrix(
+    gfx::RenderCamera& targetCamera) {
+  Magnum::Matrix4 T = this->node().absoluteTransformation();
+  Magnum::Matrix3 R = T.rotationScaling();
+  Magnum::Math::Algorithms::gramSchmidtOrthonormalizeInPlace(R);
+
+  VLOG(1) << "||R - GS(R)|| = "
+          << Eigen::Map<mat3f>((R - T.rotationShear()).data()).norm();
+
+  T = Magnum::Matrix4::from(R, T.translation()) *
+      Magnum::Matrix4::scaling(T.scaling());
+
+  // set the transformation to the camera
+  // so that the camera has the correct modelview matrix for rendering;
+  // to do it,
+  // obtain the *absolute* transformation from the sensor node,
+  // apply it as the *relative* transformation between the default camera and
+  // its parent, which is rootNode_.
+  targetCamera.node().setTransformation(T);
```

Comment:
This function has a BUG after refactoring.
The purpose is to set a "render camera" (input parameter) so that it the same "absolute" transformation as the sensor.
But the ```setTransformation``` here will only set the "relative" transformation w.r.t. the parent node. What if the "render camera" is deep in the kinematic chain?

Before, it is not a bug because it was setting the "default render camera" which attaches to the root node of the SG.

Now this new function makes it more generic by passing a regular "render camera", which causes the problem.